src/app/features/data-import/data-import.service.ts

```diff
@@ -18,12 +22,57 @@ export class DataImportService {
     private snackBar: MatSnackBar
   ) {}
 
-  async importCsvContentToDB(csv: string): Promise<void> {
-    const parsedCsv = this.papa.parse(csv, {
+  async validateCsvFile(file: File, entityType: string): Promise<CsvValidationResult> {
+    const csvData = await readFile(file);
+    const parsedCsvFile = this.parseCsvFile(csvData);
+
+    if (parsedCsvFile === undefined || parsedCsvFile.data === undefined) {
+      return {status: CsvValidationStatus.ErrorNoData, resultMessage: 'The file provided is invalid, parsing was not possible!'};
```

Comment:
Why did you decide against throwing error messages?

Reply:
@TheSlimvReal For two reasons: In most languages using "throw" is used when evaluating something that should never even happen - but is checked to make sure one can gracefully continue. So using it here to show a message box simply felt wrong. The other idea I had behind this approach is being able to use the module and functionality in other components if required, maybe like in an initial setup dialogue where you can import multiple files etc., then one would like to validate the files one after another and show another message, or no message at all. Thats why I didn't add it to the function.

But yeah, as in most other cases, I'm not that experienced in sofware engineering yet but it felt like a good idea :D